src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaHttpUtil.cs

```diff
@@ -0,0 +1,147 @@
+        private const string HeaderHost = "host";
+
+        internal static IEnumerable<KeyValuePair<string, object>> GetHttpTags<TInput>(TInput input)
+        {
+            var tags = new List<KeyValuePair<string, object>>();
+
+            string httpScheme = null;
+            string httpTarget = null;
+            string httpMethod = null;
+            string hostName = null;
+            int? hostPort = null;
+
+            switch (input)
+            {
+                case APIGatewayProxyRequest request:
+                    httpScheme = GetHeaderValue(request, HeaderXForwardedProto);
+                    httpTarget = string.Concat(request.Path ?? string.Empty, ConstructQueryString(request.QueryStringParameters));
+                    httpMethod = request.HttpMethod;
+                    var hostHeader = GetHeaderValue(request, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeader);
+                    break;
+                case APIGatewayHttpApiV2ProxyRequest requestV2:
+                    httpScheme = GetHeaderValue(requestV2, HeaderXForwardedProto);
+                    httpTarget = string.Concat(requestV2?.RequestContext?.Http?.Path ?? string.Empty, ConstructQueryString(requestV2.QueryStringParameters));
+                    httpMethod = requestV2?.RequestContext?.Http?.Method;
+                    var hostHeaderV2 = GetHeaderValue(requestV2, HeaderHost);
+                    (hostName, hostPort) = GetHostAndPort(httpScheme, hostHeaderV2);
+                    break;
+            }
+
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpScheme, httpScheme);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpTarget, httpTarget);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeHttpMethod, httpMethod);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostName, hostName);
+            tags.AddTagIfNotNull(SemanticConventions.AttributeNetHostPort, hostPort);
+
+            return tags;
+        }
+
+        internal static void SetHttpTagsFromResult(Activity activity, object result)
+        {
+            switch (result)
+            {
+                case APIGatewayProxyResponse response:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, response.StatusCode.ToString());
+                    break;
+                case APIGatewayHttpApiV2ProxyResponse responseV2:
+                    activity.SetTag(SemanticConventions.AttributeHttpStatusCode, responseV2.StatusCode.ToString());
+                    break;
+            }
+        }
+
+        internal static (string Host, int? Port) GetHostAndPort(string httpScheme, string hostHeaders)
+        {
+            if (hostHeaders == null)
+            {
+                return (null, null);
+            }
+
+            // In case of multiple headres we consider only the 1st.
+            var hostHeader = hostHeaders.Split(',').First();
```

Comment:
You could limit the split again for efficiency (split into at most 2, since we are only interested in the first). Although it would probably be even better (at least more readable) to use the multi-header getter instead, so we don't need to split at all here.